src/Equinor.ProCoSys.IPO.Domain/AggregateModels/CertificateAggregate/Certificate.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections.Generic;
+using Equinor.ProCoSys.Common;
+using Equinor.ProCoSys.Common.Time;
+using Equinor.ProCoSys.IPO.Domain.AggregateModels.InvitationAggregate;
+using Equinor.ProCoSys.IPO.Domain.AggregateModels.PersonAggregate;
+using Equinor.ProCoSys.IPO.Domain.AggregateModels.ProjectAggregate;
+using Equinor.ProCoSys.IPO.Domain.Audit;
+
+namespace Equinor.ProCoSys.IPO.Domain.AggregateModels.CertificateAggregate
+{
+    public class Certificate : PlantEntityBase, IAggregateRoot, ICreationAuditable
+    {
+        private readonly List<McPkg> _certificateMcPkgScope = new List<McPkg>();
+        private readonly List<CommPkg> _certificateCommPkgScope = new List<CommPkg>();
+
+        protected Certificate()
+            : base(null)
+        {
+        }
+
+        public Certificate(
+            string plant,
+            Project project,
+            Guid pcsGuid)
+            : base(plant)
+        {
+            if (project is null)
+            {
+                throw new ArgumentNullException(nameof(project));
+            }
+          
```

Comment:
throw exception if plant != project.Plant

Reply:
I see that this check misses other places too ... sample: invitation ctor.
Sample code where check is ok is invitation.AddComment

Reply:
I can't see that such code can be done there

Reply:
Ups, confused with checking if plant is empty/null